flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/FlinkLogicalRankRule.scala

```diff
@@ -98,6 +98,12 @@ abstract class FlinkLogicalRankRuleBase
       case _ => throw new TableException(s"Unsupported rank function: $rankFun")
     }
 
+    // this is to avoid duplicated column name with the input
+    var rankNumberName = rankNumberType.get.getName
+    while (window.getInput.getRowType.getFieldNames.contains(rankNumberName)) {
+      rankNumberName += "$"
```

Comment:
When hitting duplicate field names, Calcite ususally append `_<id>` after it, e.g. `$g_0`, `$g_1`. 
What do you think about `rankNumberName += "_0"`?